src/rpiplatesrecognition/forms.py

```diff
@@ -65,6 +65,18 @@ def validate_whitelist_name(self, whitelist_name):
         if whitelist is not None:
             raise ValidationError('This name is already taken')
         
-        if len(whitelist_name.data) < 2:
-            raise ValidationError('Too short name')
+class AddPlateForm(FlaskForm):
 
+    licence_plate_number = StringField('Licence plate number', validators=[DataRequired(),Length(4,10)])
+    submit = SubmitField('Add licence plate')
+
+    def __init__(self, whitelist_id: int, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.whitelist_id = whitelist_id
+
+    def validate_licence_plate_number(self, licence_plate_number):
+        whitelist = Whitelist.query.filter_by(id = self.whitelist_id).first()
+        licence_plate = Plate.query.filter_by(text=licence_plate_number.data).first()
```

Comment:
There's a possibility that there's no such license plate in db (as it needs to be created)

add an `if license_plate is not None` statement